File: DataStruct_Algorithm/Permutation_and_Combination/combination.php
<?php
/***************************************************************************
 * 
 * Copyright (c) 2015 Baidu.com, Inc. All Rights Reserved
 * 
 **************************************************************************/

function Combination($arr, $idx, $k, $got_list, &$res) {
    if($k == 0) {
        $res []= $got_list;
        return;
    }

    for($i = $idx; $i < count($arr); $i++) {
		$got_list[$k] = $arr[$i];
        Combination($arr, $i + 1, $k - 1, $got_list, $res);
    }
}


$arr = array(1, 3, 5, 7, 9);
$arr_size = count($arr);

$res = array();
Combination($arr, 0, 3, array(), $res);

foreach($res as $v) {
	foreach($v as $v1) {
		echo $v1;
	}
	echo "\n";
}

File: DataStruct_Algorithm/Permutation_and_Combination/combination.cpp
/***************************************************************************
 * 
 * Copyright (c) 2015 Baidu.com, Inc. All Rights Reserved
 * 
 **************************************************************************/

#include <iostream>
#include <vector>

using namespace std;

void getCombination(const vector<int>& arr, int cur, vector<int> got_arr, int size, vector<vector<int> >& res) {
    if(got_arr.size() == size) {
        res.push_back(got_arr);
        return;
    }
    if(cur >= arr.size()) {
        return;
    }
    
    if(arr.size() - cur >= size - got_arr.size()) {
        getCombination(arr, cur + 1, got_arr, size, res);
        got_arr.push_back(arr[cur]);
        getCombination(arr, cur + 1, got_arr, size, res);
    }
}

int main() {
    vector<vector<int> > res;

    int arr[] = {1, 2, 3, 4, 5};
    vector<int> v(arr, arr + sizeof(arr)/sizeof(int));

    for(int i = 1; i <= v.size(); i++) {
        vector<int> tmp;
        getCombination(v, 0, tmp, i, res);
    }

    vector<vector<int> >::iterator it1;
    for(it1 = res.begin(); it1 != res.end(); it1++) {
        vector<int>::iterator it2;
        for(it2 = it1->begin(); it2 != it1->end(); it2++) {
            cout << *it2;
        }
        cout << endl;
    }
    
    return 0;
}
